activerecord/test/cases/finder_test.rb

```diff
@@ -22,13 +22,13 @@ class FinderTest < ActiveRecord::TestCase
   fixtures :companies, :topics, :entrants, :developers, :developers_projects, :posts, :comments, :accounts, :authors, :author_addresses, :customers, :categories, :categorizations, :cars
 
   def test_find_by_id_with_hash
-    assert_raises(ActiveRecord::StatementInvalid) do
+    assert_nothing_raised do
```

Comment:
Those tests are by design and I think it is better behavior. If `id` is a integer column we should not allow it to query by a hash. What is the generated SQL in this case?